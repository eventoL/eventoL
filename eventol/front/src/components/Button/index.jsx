import React from 'react';
import PropTypes from 'prop-types';

export default class Button extends React.PureComponent {
  static propTypes = {
    handleOnClick: PropTypes.func.isRequired,
    label: PropTypes.string,
    name: PropTypes.string,
    type: PropTypes.oneOf([
      'default',
      'primary',
      'success',
      'info',
      'warning',
      'danger',
      'link',
    ]),
  };

  static defaultProps = {
    label: '',
    name: 'btn',
    type: 'default',
  };

  handleClick = event => {
    event.preventDefault();
    event.stopPropagation();
    const {handleOnClick} = this.props;
    handleOnClick(event.target.id);
  };

  render() {
    const {type, label, name} = this.props;
    const className = `btn btn-raised btn-${type}`;
    return (
      <button
        className={className}
        id={name}
        onClick={this.handleClick}
        type="button"
      >
        <span aria-hidden id={name}>
          {label}
        </span>
      </button>
    );
  }
}
