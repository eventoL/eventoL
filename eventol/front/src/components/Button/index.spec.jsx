import React from 'react';
import renderer from 'react-test-renderer';

import Button from '.';

describe('Button', () => {
  let component;
  let tree;
  let onClickSpy;
  let event;

  beforeEach(() => {
    onClickSpy = jest.fn();
    event = {
      preventDefault: jest.fn(),
      stopPropagation: jest.fn(),
      target: {
        id: 'button',
      },
    };
    component = renderer.create(<Button handleOnClick={onClickSpy} />);
    tree = component.toJSON();
  });

  test('Snapshot', () => {
    expect(tree).toMatchSnapshot();
  });

  test('should call onClick handle', () => {
    tree.props.onClick(event);
    expect(onClickSpy).toBeCalled();
    expect(onClickSpy).toBeCalledWith('button');
  });
});
