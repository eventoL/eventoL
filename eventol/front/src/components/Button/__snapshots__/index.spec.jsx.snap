// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot 1`] = `
<button
  className="btn btn-raised btn-default"
  id="btn"
  onClick={[Function]}
  type="button"
>
  <span
    aria-hidden={true}
    id="btn"
  >
    
  </span>
</button>
`;
